### PR TITLE
[WIP] codes_claimed by issuer

### PR DESCRIPTION
**Pull request by whaught**
## Proposed Changes

* This feature is experimental and requires privacy review & potentially checks for  a minimum group size to prevent de-anonymized data.
* This PR is just looking at what it takes to add these fields

### CODE REVIEW

File: pkg/database/token.go [hunk 1]
```diff
@@ -414,7 +414,46 @@ func (db *Database) updateStatsCodeClaimed(t time.Time, authApp *AuthorizedApp)
...
+				ON CONFLICT (date, realm_id, issuer_id) DO UPDATE
+					SET codes_claimed = external_issuer_stats.codes_claimed + 1`
+
+		if err := db.db.Exec(sql, midnight, vc.RealmID, vc.IssuingExternalID).Error; err != nil {
+			db.logger.Warnw("failed to update external-issuer stats", "error", err)
+		}
+	}
+
+	// Update the per-realm stats.
+	if vc.RealmID != 0 {
```
Comment (whaught) [inline pkg/database/token.go:447]: This should already be happening somewhere, but this is what we do for `realm_stats.tokens_claimed` and `realm_stats.codes_issued`

Comment (whaught) [reply pkg/database/token.go]: I grepped for it wrong. we do CodesClaimed++ below

File: pkg/database/realm.go [hunk 2]
```diff
@@ -1597,6 +1597,7 @@ func (r *Realm) ExternalIssuerStats(db *Database) (ExternalIssuerStats, error) {
 			$1 AS realm_id,
 			d.issuer_id AS issuer_id,
 			COALESCE(s.codes_issued, 0) AS codes_issued
```
Comment (sethvargo) [inline pkg/database/realm.go:1599]: Need comma

**Review by sethvargo — COMMENTED** [review]
Okay, so it's not _terrible_

File: pkg/database/realm.go (hunk 2 above)
Comment (whaught) [reply pkg/database/realm.go]: which means it also needs a test... looking at that too

**Pull request closed without merging**